lib/msf/java/rmi/client/jmx/connection.rb

```diff
@@ -0,0 +1,122 @@
+# -*- coding: binary -*-
+
+module Msf
+  module Java
+    module Rmi
+      module Client
+        module Jmx
+          # This mixin provides methods to simulate calls to the Java
+          # javax/management/remote/rmi/RMIConnectionImpl_Stub
+          # interface
+          module Connection
+            require 'msf/java/rmi/client/jmx/connection/builder'
+
+            include Msf::Java::Rmi::Client::Jmx::Connection::Builder
+
+            # Sends a call to the JMXRMI endpoint to retrieve an MBean instance. Simulates a call
+            # to the Java javax/management/remote/rmi/RMIConnectionImpl_Stub#getObjectInstance()
+            # method.
+            #
+            # @param opts [Hash]
+            # @option opts [Rex::Socket::Tcp] :sock
+            # @return [TrueClass, NilClass] true if success, nil otherwise
+            # @raise [Rex::Proto::Rmi::Exception] if the endpoint raises a remote exception
+            # @see Msf::Java::Rmi::Client::Registry::Builder.build_jmx_get_object_instance
+            def send_jmx_get_object_instance(opts = {})
+              send_call(
+                sock: opts[:sock] || sock,
+                call: build_jmx_get_object_instance(opts)
+              )
+
+              return_value = recv_return(
+                sock: opts[:sock] || sock
+              )
+
+              if return_value.nil?
+                return nil
+              end
+
+              if return_value.is_exception?
```

Comment:
In fact, this whole method looks like copy-pasta. Perhaps `recv_return` could take an `expected_class_name` argument and raise or return `nil` if that's not what we got.  e.g.:
```ruby
def send_jmx_get_object_insance(opts)
  my_sock = opts[:sock] || sock
  send_call(sock: my_sock, call: build_jmx_get_object_instance(opts))
    
  return_value = recv_return(
    sock: my_sock,
    expected_class_name: 'javax.management.ObjectInstance',
  )
  return !!return_value
end